File: connectorSamples/xyz-airquality/src/XyzResponse.py
class XyzResponse:
    def __init__(self, responseType):
        self.type = responseType


class SuccessResponse(XyzResponse):
    def __init__(self, status):
        super().__init__("SuccessResponse")
        self.status = status


class ErrorResponse(XyzResponse):
    def __init__(self, streamId=None, error=None, errorMessage=None):
        super().__init__("ErrorResponse")
        self.error = error
        self.errorMessage = errorMessage
        self.streamId = streamId


class FeatureCollection(XyzResponse):
    def __init__(self, features, etag=None, handle=None):
        super().__init__("FeatureCollection")
        self.features = features
        if etag:
            self.etag = etag
        if handle:
            self.handle = handle


class StatisticsResponse(XyzResponse):
    def __init__(self, bboxValue, byteSizeValue, countValue, geometryTypes, propertiesArray):
        super().__init__("StatisticsResponse")
        self.bbox = {"value": bboxValue, "estimated": False}
        self.byteSize = {"value": byteSizeValue, "estimated": False}
        self.count = {"value": countValue, "estimated": False}
        self.geometryTypes = {"value": geometryTypes, "estimated": False}
        self.properties = {
            "value": propertiesArray,
            "estimated": False,
            "searchable": "PARTIAL"
        }
        self.tags = {
            "value": [],
            "estimated": False
        }

    def add_tags(self, key, count):
        self.tags["value"].append({
            "key": key,
            "count": count
        })

File: connectorSamples/xyz-wfs-typescript/src/interfaces/XyzResponse.ts
import {Feature} from "geojson";

export class XyzResponse {
    type: string;
    features?: Feature[];
    errorMessage?: string;
    constructor() {
        this.type = this.constructor.name
    }
}

export class SuccessResponse extends XyzResponse{
    status: string;
    constructor() {
        super();
        this.status = "OK";
    }
}

export class ErrorResponse extends XyzResponse {
    streamId: string;
    error: number;
    errorMessage: string;
    constructor(streamId, code, message) {
        super();
        this.streamId = streamId;
        this.error = code;
        this.errorMessage = message;
    }
}

export class HealthResponse extends SuccessResponse {
    constructor(status) {
        super();
        this.status = status;
        this.type = "HealthStatus";
    }
}

export class FeatureCollection extends XyzResponse{
    type: string = "FeatureCollection";
    features?: any[];
    etag?: string;
    handle?: any;

    constructor(features = []) {
        super();
        this.features = features;
    }
}

export interface StatisticValue {
    value: number | string | string[] | number[] | object[]
    estimated?: boolean
    searchable?: string
}

export interface TagValue {
    value: object[],
    estimated?: boolean
}

export interface GeometryValue {
    value: string[],
    estimated: boolean
}

export class StatisticsResponse extends FeatureCollection {
    type: string;
    count: StatisticValue;
    byteSize: StatisticValue;
    bbox: StatisticValue;
    geometryTypes: GeometryValue;
    properties: StatisticValue;
    tags: TagValue;
}

export const emptyStatisticsResponse :StatisticsResponse = {
    type: "StatisticsResponse",
    count: {
        "value": 0,
        "estimated": true
    },
    byteSize: {
        "value": 0,
        "estimated": true
    },
    bbox: {
        "value": [],
        "estimated": true
    },
    geometryTypes: {
        "value": [],
        "estimated": true
    },
    properties: {
        "value": [],
        "estimated": true,
        "searchable": "PARTIAL"
    },
    tags: {
        "value": []
    }
};
